### PR TITLE
Add pedantic and fix numerous pedantic errors

### PR DESCRIPTION
**Pull request by lanza**
lldb uses -Wpedantic and thus incorporating ds2 builds into lldb
requires building with pedantry

### CODE REVIEW

File: Sources/Utils/Stringify.cpp [hunk 1]
```diff
@@ -12,6 +12,9 @@
 #include "DebugServer2/Utils/Stringify.h"
 #include "DebugServer2/Utils/Log.h"
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wcovered-switch-default"
```
Comment (xiaobai) [inline Sources/Utils/Stringify.cpp]: This breaks the project. :(

File: Sources/Core/ARM/HardwareBreakpointManager.cpp [hunk 2]
```diff
@@ -19,17 +19,17 @@ namespace ds2 {
 
 int HardwareBreakpointManager::hit(Target::Thread *thread, Site &site) {
   return -1;
-};
```
Comment (sas) [inline Sources/Core/ARM/HardwareBreakpointManager.cpp]: The changes in this file are good. I'm not sure wtf these semicolons are doing there...

Comment (sas) [inline Sources/Core/ARM/HardwareBreakpointManager.cpp]: The changes in this file are good. I'm not sure wtf these semicolons are doing there...

File: Sources/GDBRemote/DebugSessionImpl.cpp [hunk 3]
```diff
@@ -206,9 +206,6 @@ ErrorCode DebugSessionImplBase::queryStopInfo(Session &session, Thread *thread,
   case StopInfo::kEventKill:
     DS2ASSERT(stop.reason == StopInfo::kReasonNone);
     break;
-
```
Comment (sas) [inline Sources/GDBRemote/DebugSessionImpl.cpp]: This is fine too, I guess.

File: Sources/GDBRemote/DummySessionDelegateImpl.cpp [hunk 4]
```diff
@@ -31,7 +31,7 @@ namespace GDBRemote {
 DummySessionDelegateImpl::DummySessionDelegateImpl() = default;
 
 size_t DummySessionDelegateImpl::getGPRSize() const {
-  DS2ASSERT(!"this method shall be implemented by inheriting class");
```
Comment (sas) [inline Sources/GDBRemote/DummySessionDelegateImpl.cpp]: Instead of using `false`, you should make this a `DS2BUG()`.

**Pull request closed without merging**